### ISSUE TITLE
NSGenericException: Collection was mutated while being enumerated

### ISSUE THREAD

**Issue by NickFitton** [bugsnag/bugsnag-js#1830]
### Describe the bug
An uncommon unexpected error occurs that seems to be a race condition in adding breadcrumbs.
I am aware that this is a very outdated version, we hope to upgrade in our next native release and if that resolves the problem I will close the issue with that as a solution.

The bugsnag parts of the stacktrace refer to:
```
- (NSArray *)collectBreadcrumbs {
    NSMutableArray *crumbs = self.breadcrumbs.breadcrumbs;
    NSMutableArray *data = [NSMutableArray new];

    for (BugsnagBreadcrumb *crumb in crumbs) {                              < BugsnagClient.m:1569:5
        NSMutableDictionary *crumbData = [[crumb objectValue] mutableCopy];
        // JSON is serialized as 'name', we want as 'message' when passing to RN
        crumbData[@"message"] = crumbData[@"name"];
        crumbData[@"name"] = nil;
        crumbData[@"metadata"] = crumbData[@"metaData"];
        crumbData[@"metaData"] = nil;
        [data addObject: crumbData];
    }
    return data;
}
```
and
```
RCT_EXPORT_METHOD(getPayloadInfo:(NSDictionary *)options
                         resolve:(RCTPromiseResolveBlock)resolve
                          reject:(RCTPromiseRejectBlock)reject) {
    BugsnagClient *client = [Bugsnag client];
    NSMutableDictionary *info = [NSMutableDictionary new];
    info[@"app"] = [client collectAppWithState];
    info[@"device"] = [client collectDeviceWithState];
    info[@"breadcrumbs"] = [client collectBreadcrumbs];                    < BugsnagReactNative.m:139:28
    BOOL unhandled = [options[@"unhandled"] boolValue];
    info[@"threads"] = [client collectThreads:unhandled];
    resolve(info);
}
```

Error looks similar to https://github.com/bugsnag/bugsnag-js/issues/1042 just in a different place.

### Steps to reproduce
Unsure, but seems to be triggered rarely when adding a breadcrumb (142 occurances in 30 days with 3m sessions).

### Environment
* Bugsnag version: `"@bugsnag/react-native": "^7.4.0"`
* Device (e.g. iphonex): iOS only but seems to affect all models (36 are listed in bugsnag)

<details><summary>Error messages:</summary>

```
CrashReporter Key:  fa455eae983fe133614462ee9a86fff2f278b1e4
Hardware Model:     iPhone13,3
Process:            Our app
Identifier:         Our.app
Version:            Our app version
Role:               Foreground
OS Version:         iOS 15.6


NSGenericException: *** Collection <__NSArrayM: 0x282c1b6c0> was mutated while being enumerated.

0  CoreFoundation          ___exceptionPreprocess
1  libobjc.A.dylib         _objc_exception_throw
2  CoreFoundation          ___NSFastEnumerationMutationHandler
3  Our app              -[BugsnagClient collectBreadcrumbs] (BugsnagClient.m:1569:5)
4  Our app              -[BugsnagReactNative getPayloadInfo:resolve:reject:] (BugsnagReactNative.m:139:28)
5  CoreFoundation          ___invoking___
6  CoreFoundation          -[NSInvocation invoke]
7  CoreFoundation          -[NSInvocation invokeWithTarget:]
8  Our app              -[RCTModuleMethod invokeWithBridge:module:arguments:] (RCTModuleMethod.mm:584:3)
9  Our app              facebook::react::invokeInner(RCTBridge*, RCTModuleData*, unsigned int, folly::dynamic const&, int, (anonymous namespace)::SchedulingContext) (RCTNativeModule.mm:183:17)
10 Our app              facebook::react::RCTNativeModule::invoke(unsigned int, folly::dynamic&&, int)::$_0::operator()() const (RCTNativeModule.mm:104:7)
11 libdispatch.dylib       __dispatch_call_block_and_release
12 libdispatch.dylib       __dispatch_client_callout
13 libdispatch.dylib       __dispatch_lane_serial_drain
14 libdispatch.dylib       __dispatch_lane_invoke
15 libdispatch.dylib       __dispatch_workloop_worker_thread
16 libsystem_pthread.dylib __pthread_wqthread
17 libsystem_pthread.dylib _start_wqthread
```
</details>

**Comment by nickdowell** [comment]
Just FYI the underlying fix was part of [bugsnag-cocoa v6.2.3](https://github.com/bugsnag/bugsnag-cocoa/releases/tag/v6.2.3) and the first bugsnag-js release to include that was [v7.5.2](https://github.com/bugsnag/bugsnag-js/releases/tag/v7.5.2)

**Comment by johnkiely1** [comment]
@NickFitton 

I'm going to close this out now on the basis that the upgrade should solve the issue. Let us know if you see anything different.
